packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -172,10 +172,10 @@ class BottomSheet extends StatefulWidget {
   /// This API available as a convenience for a Material compliant bottom sheet
   /// animation. If alternative animation durations are required, a different
   /// animation controller could be provided.
```

Comment:
Hi @ZainUrRehmanKhan It looks like you can already provide your own `AnimationController` to the `BottomSheet` (`BottomSheet.animationController`), with which you can specify your own durations.

I am not sure that this API make sense, what happens if a user specifies an AnimationController and and enterDuration/exitDuration? How would that be reconciled?

Reply:
Hi @Piinks, yes we can provide `AnimationController` to `BottomSheet `  but it's not sufficient. What if we need custom animation for `showModalBottomSheet/showBottomSheet`, we can't provide `AnimationController` there,  I faced this issue so many times where I have to set my custom `Duration`s.

> I am not sure that this API make sense, what happens if a user specifies an AnimationController and enterDuration/exitDuration? How would that be reconciled?

I haven't added these properties `enterDuration/exitDuration` to BottomSheet because we can provide `AnimationController` there, so there will be no collision between AnimationController and `enterDuration/exitDuration`. These properties `enterDuration/exitDuration` are just added to `showModalBottomSheet/showBottomSheet` so we can use our own `Duration`s which override the default `_bottomSheetEnterDuration/_bottomSheetExitDuration` here:  

 ```
 static AnimationController createAnimationController(TickerProvider vsync, {Duration? enterDuration, Duration? 
   exitDuration}) {
      return AnimationController(
        duration: enterDuration ?? _bottomSheetEnterDuration,
        reverseDuration: exitDuration ?? _bottomSheetExitDuration,
        debugLabel: 'BottomSheet',
        vsync: vsync,
    );
  }
```

The documentation you have mentioned if of `BottomSheet `, I'm talking about   `showModalBottomSheet/showBottomSheet`
> /// animation. If alternative animation durations are required, a different
  /// animation controller could be provided.


**I also tried this, but this not work,**
```
showModalBottomSheet(
                context: context,
                builder: (context) {
                  return BottomSheet(
                    animationController: AnimationController(duration: Duration(seconds: 5), vsync: this),
                    builder: (context) {
                      return Container(child: Text('hello'), height: 100,);
                    },
                    onClosing: () {

                    },
                  );
                },
              );
```

**so I think adding `enterDuration/exitDuration` is much a better, easy, and reliable option:**
```
showModalBottomSheet(
                context: context,
                builder: (context) {
                  return Container(height: 100, child: Text('hello world'),);
                },
                enterDuration: Duration(seconds: 3),
                endDuration: Duration(seconds: 3)
              );
```

Reply:
?

Reply:
Could you clarify why passing an animationController into `showModalBottomSheet` and `showBottomSheet` would not work? You could try creating an API as follows:
```dart
showModalBottomSheet(
  context: context,
  builder: (context) {
    return Container(height: 100, child: Text('hello world'),);
  },
  transitionAnimationController: AnimationController(
    duration: Duration(seconds: 5), 
    vsync: Navigator.of(context).overlay,
  ),
);
```

Then, basically what you can do is say if an animation controller is passed into the `show*` APIs, to use that animation controller instead of creating a new one.

Reply:
That being said, I'm not opposed to just passing the durations in as well -- that is in fact what I did in the animations package (see [example here](https://github.com/flutter/packages/pull/60/files#diff-bac91d95af5e180d3094637cec2bcd3f3def7894c1ae86e74f8082351d4056c1R67)). Passing an animation controller does seem cleaner.

Reply:
@shihaohong yeah! you are right. Thanks for the suggestion, I'll revert my previous changes and add an option for passing animationController into `showModalBottomSheet` and `showBottomSheet`.